pandas/core/internals/blocks.py

```diff
@@ -944,15 +944,14 @@ def putmask(self, mask, new, align=True, inplace=False, axis=0, transpose=False)
                 and np.any(mask[mask])
                 and getattr(new, "ndim", 1) == 1
             ):
-
-                if not (
-                    mask.shape[-1] == len(new)
-                    or mask[mask].shape[-1] == len(new)
-                    or len(new) == 1
-                ):
+                if mask[mask].shape[-1] == len(new):  # GH 30567
```

Comment:
can you move the comment to the next line and add a sentence explaining the issue

Reply:
done